src/util/sfdcUrl.ts

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { URL } from 'url';
+import { Env, Duration } from '@salesforce/kit';
+import { MyDomainResolver } from '../status/myDomainResolver';
+import cache from './cache';
+
+export class SfdcUrl extends URL {
+  /**
+   * Salesforce URLs.
+   */
+  public static SANDBOX = 'https://test.salesforce.com';
+  public static PRODUCTION = 'https://login.salesforce.com';
+
+  public constructor(input: string | URL, base?: string | URL) {
+    super(input.toString(), base);
+    if (this.protocol !== 'https:' && !cache.has(this.origin)) {
+      cache.set(this.origin, {
+        timestamp: new Date().getTime(),
+      });
+      this.emitWarning('Using insecure protocol: ' + this.protocol + ' on url: ' + this.origin);
+    }
+  }
+
+  /**
+   * Returns the appropiate jwt audience url for this url.
+   */
+  public async getJwtAudienceUrl(createdOrgInstance?: string): Promise<string> {
+    // environment variable is used as an override
+    if (process.env.SFDX_AUDIENCE_URL) {
+      return process.env.SFDX_AUDIENCE_URL;
```

Comment:
We might want to consider adding debug output here that the audience URL is being overridden by the env var, and what the value of the env var is.

Reply:
Using Logger class to debug overriden value